auditors/hostns/hostns.go

```diff
@@ -0,0 +1,122 @@
+package hostns
+
+import (
+	"github.com/Shopify/kubeaudit"
+	"github.com/Shopify/kubeaudit/internal/k8s"
+	"github.com/Shopify/kubeaudit/internal/override"
+	"github.com/Shopify/kubeaudit/k8stypes"
+)
+
+// TODO this check used to only work on Pod resources, but now it works on any resource that has pods (including
```

Comment:
We need to find out :) I didn't write the initial code @jonpulsifer might have the answer we are looking for.

Reply:
I can't follow the blame for this so I don't know the history :joy:

If I can make a guess, I naively implemented the checks by just selecting all the pods regardless of workload controller

I think the initial reason for this was pod security policies, for example:

Here is the yaml output of `kubectl get deploy/cert-manager -n cert-manager` in my cluster using PSPs

```yaml
# kubectl get deploy/cert-manager -n cert-manager
spec:
      containers:
      - args:
        - --v=2
        - --cluster-resource-namespace=$(POD_NAMESPACE)
        - --leader-election-namespace=kube-system
        - --webhook-namespace=$(POD_NAMESPACE)
        - --webhook-ca-secret=cert-manager-webhook-ca
        - --webhook-serving-secret=cert-manager-webhook-tls
        - --webhook-dns-names=cert-manager-webhook,cert-manager-webhook.cert-manager,cert-manager-webhook.cert-manager.svc
        env:
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        image: quay.io/jetstack/cert-manager-controller:v0.11.0
        imagePullPolicy: IfNotPresent
        name: cert-manager
        ports:
        - containerPort: 9402
          protocol: TCP
        resources:
          requests:
            cpu: 10m
            memory: 32Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
```

Here is the output of the only replica in that deployment:

```yaml
# kubectl get pod/cert-manager-fc49fbbc5-zrcpq -o yaml
spec:
  containers:
  - args:
    - --v=2
    - --cluster-resource-namespace=$(POD_NAMESPACE)
    - --leader-election-namespace=kube-system
    - --webhook-namespace=$(POD_NAMESPACE)
    - --webhook-ca-secret=cert-manager-webhook-ca
    - --webhook-serving-secret=cert-manager-webhook-tls
    - --webhook-dns-names=cert-manager-webhook,cert-manager-webhook.cert-manager,cert-manager-webhook.cert-manager.svc
    env:
    - name: POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    image: quay.io/jetstack/cert-manager-controller:v0.11.0
    imagePullPolicy: IfNotPresent
    name: cert-manager
    ports:
    - containerPort: 9402
      protocol: TCP
    resources:
      requests:
        cpu: 10m
        memory: 32Mi
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - AUDIT_CONTROL
        - AUDIT_WRITE
        - BLOCK_SUSPEND
        - CHOWN
        - DAC_OVERRIDE
        - DAC_READ_SEARCH
        - FOWNER
        - FSETID
        - IPC_LOCK
        - IPC_OWNER
        - KILL
        - LEASE
        - LINUX_IMMUTABLE
        - MAC_ADMIN
        - MAC_OVERRIDE
        - MKNOD
        - NET_ADMIN
        - NET_BROADCAST
        - NET_RAW
        - SETFCAP
        - SETGID
        - SETPCAP
        - SETUID
        - SYSLOG
        - SYS_ADMIN
        - SYS_BOOT
        - SYS_CHROOT
        - SYS_MODULE
        - SYS_NICE
        - SYS_PACCT
        - SYS_PTRACE
        - SYS_RAWIO
        - SYS_RESOURCE
        - SYS_TIME
        - SYS_TTY_CONFIG
        - WAKE_ALARM
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: cert-manager-token-t7m84
      readOnly: true
```

Reply:
Ok cool so it only ran on Pods before because we only cared about Pods at the time, not because it *shouldn't* run on other resources. I will remove the TODO in a future PR.

For reference, here is the "history" https://github.com/Shopify/kubeaudit/blob/9158be4230eb480604ac06040c8277c59bb9ba7c/cmd/namespaces.go#L101